internal/thelma/state/providers/sherlock/app_release.go

```diff
@@ -0,0 +1,62 @@
+package sherlock
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/broadinstitute/thelma/internal/thelma/state/api/terra"
+)
+
+type appRelease struct {
+	appVersion string
+	subdomain  string
+	protocol   string
+	port       int
+	release
+}
```

Comment:
I totally get that we're just trying to match Thelma's existing interfaces here, not asking for changes, more wondering if down the road we plan to eliminate the app-release-versus-cluster-release distinction. 

Sherlock and Beehive don't really make a distinction there--they'll by all means let you work with the app version field of a "cluster release" even though it wouldn't end up being read by Thelma. I suspect we're the only ones that would come across that weird behavior--I'm perfectly cool with the current state of things if the goal is to combine these release types down the road. I think it would be fairly easy once we're off helmfile because Sherlock provides these fields for all releases.

Reply:
Yeah I was going to combine them both fully down the road, they are already partially combined. Both are already calling the same state constructor under the hood in the sherlock/client package. I realize there is no distinction between the two at the sherlock level. This was mainly a method for me to figure out how Thelma's existing stateloading mechanism works.

Reply:
I was planning to a fair amount of consolidation in my next pass